src/output.rs

```diff
@@ -0,0 +1,82 @@
+// commands.rs
+
+// Copyright (C) 2020 The Nitrocli Developers
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+use crate::Context;
+
+/// A progress bar that can be printed to an interactive output.
+pub struct ProgressBar {
+  redraw: bool,
+  progress: u8,
+  toggle: bool,
+  finished: bool,
```

Comment:
Would you mind documenting the members briefly? I am somehow failing to grasp the purpose of `toggle`.

Reply:
Sure.  `toggle` is just used to display a pulsing indicator (`.` -> ` ` -> `.`) to show that the process is not stuck.